internal/ice/util.go

```diff
@@ -12,28 +12,39 @@ func localInterfaces() (ips []net.IP) {
 	}
 
 	for _, iface := range ifaces {
-		if iface.Flags&net.FlagUp == 0 {
-			continue // interface down
-		}
-		if iface.Flags&net.FlagLoopback != 0 {
-			continue // loopback interface
+		if iface.Flags&net.FlagUp == 0 || iface.Flags&net.FlagLoopback != 0 {
+			continue
 		}
+
 		addrs, err := iface.Addrs()
 		if err != nil {
 			return ips
 		}
+
 		for _, addr := range addrs {
 			var ip net.IP
-			switch v := addr.(type) {
+			switch addr := addr.(type) {
 			case *net.IPNet:
-				ip = v.IP
+				ip = addr.IP
 			case *net.IPAddr:
-				ip = v.IP
+				ip = addr.IP
+
 			}
+
 			if ip == nil || ip.IsLoopback() {
 				continue
 			}
-			ips = append(ips, ip)
+
+			// The conditions of invalidation written below are defined in
+			// https://tools.ietf.org/html/rfc8445#section-5.1.1.1
```

Comment:
can you keep the append at the end, and do early returns like the preceding code just to stay consistent